lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -169,6 +169,35 @@ static Value createSoftmaxBackwardCommonKernel(PatternRewriter &rewriter,
   return sub;
 }
 
+// This algorithm is adopted from:
+// https://github.com/pytorch/pytorch/blob/c47bdd7522c7146c091c95ff5b4ce354d7a8870c/aten/src/ATen/native/TensorShape.cpp#L3660
+static SmallVector<int64_t>
+computeDimsOrderForMoveDim(ArrayRef<int64_t> srcDimInt,
+                           ArrayRef<int64_t> dstDimInt, unsigned inputRank) {
```

Comment:
I don't see a need for `srcDimInt` and `dstDimInt` to be arrays. They can just be `int`s